lib/usb/usb_standard.c

```diff
@@ -268,32 +275,53 @@ static int usb_standard_set_interface(usbd_device *usbd_dev,
 				      struct usb_setup_data *req,
 				      uint8_t **buf, uint16_t *len)
 {
-	(void)usbd_dev;
-	(void)req;
+	const struct usb_config_descriptor * cfx = &usbd_dev->config[usbd_dev->current_config - 1];
+	const struct usb_interface * iface;
+
 	(void)buf;
 
-	/* FIXME: Adapt if we have more than one interface. */
-	if (req->wValue != 0) {
-		return 0;
+	if (req->wIndex >= cfx->bNumInterfaces) {
+		return USBD_REQ_NOTSUPP;
 	}
+
+	iface = &cfx->interface[req->wIndex];
+	
+	if (req->wValue >= iface->num_altsetting) {
+		return USBD_REQ_NOTSUPP;
+	}
+
+	if(iface->cur_altsetting) {
```

Comment:
space after the if and before the ( here please (same below in a couple of places)